app/_components/TestComponent/BadComponent.tsx

```
@@ -0,0 +1,93 @@
+                Click me (Counter: {counter})
+            </button>
+
+            {/* Bad: Rendering large arrays directly in JSX without memoization */}
+            {dataList.map((item) => (
+                <div key={item.id} onClick={() => inlineFunc(item.id)}>
+                    {item.name}
+                </div>
+            ))}
+
+            {/* Very bad: dangerouslySetInnerHTML */}
+            <div dangerouslySetInnerHTML={{ __html: htmlString }}></div>
```

Biome (1.9.4)

[error] Avoid passing content using the dangerouslySetInnerHTML prop.

Setting content using code can expose users to cross-site scripting (XSS) attacks

(lint/security/noDangerouslySetInnerHtml)